src/hotspot/share/logging/logConfiguration.cpp

```diff
@@ -211,6 +211,21 @@ void LogConfiguration::delete_output(size_t idx) {
   delete output;
 }
 
+// MT-SAFETY
+//
+// ConfigurationLock can guarantee that only one thread is performing reconfiguration. This function still needs
+// to be MT-safe because logsites in other threads may be executing in parallel. Reconfiguration means unified
+// logging allows users to dynamically change tags and decorators of a log output via DCMD(logDiagnosticCommand.hpp).
+//
+// A RCU-style synchronization 'wait_until_no_readers()' is imposed inside of 'ts->set_output_level(output, level)'
+// if setting has changed. It guarantees that all logs either synchronous writing or enqueuing to the async buffer
```

Comment:
// if a setting has changed. It guarantees that all logs, either synchronous writes or enqueuing to the async buffer,